lesson_04/script.js

```diff
@@ -0,0 +1,20 @@
+"use strict";
+const ul = document.getElementById("js-target");
+const fragment = document.createDocumentFragment();
+const data = [
+  { to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク" },
+  { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" }
+];
+for (let i = 0; i <= data.length - 1; i++) {
```

Comment:
Q
Why did you set data.length - 1?

Reply:
Because it could not function without the -1.
My thought was to add -1 because the for statement would start from 0.

Reply:
```suggestion
for (let i = 0; i < data.length; i++) {
```
You can use `i<` instead of `i <=`. `<=` would have resulted in `i` being greater than or equal to the number of `i`'s, which would have prevented the length from being obtained.